service/history/historyEngine.go

```diff
@@ -1064,13 +1083,19 @@ Update_History_Loop:
 		}
 
 		var transferTasks []persistence.Task
+		var timerTasks []persistence.Task
 		if !msBuilder.HasPendingDecisionTask() {
 			newDecisionEvent, _ := msBuilder.AddDecisionTaskScheduledEvent()
 			transferTasks = []persistence.Task{&persistence.DecisionTask{
 				DomainID:   domainID,
 				TaskList:   *newDecisionEvent.DecisionTaskScheduledEventAttributes.TaskList.Name,
 				ScheduleID: *newDecisionEvent.EventId,
 			}}
+			if len(msBuilder.executionInfo.StickyTaskList) > 0 {
```

Comment:
Looks like this is repeated multiple times, can you add a helper method?

Reply:
added mutableStateBuilder.isStickyTaskListEnabled()